### PR TITLE
refactor: implement multi select interface

### PR DESCRIPTION
**Pull request by sissbruecker**
## Description

Implements the `MultiSelect` interface for the multi select combo box. Selection logic has been extracted into a `MultiSelectComboBoxSelectionModel`, similar to the selection models of the grid. 
The selection model keeps a redundant value / selection state on top of the component's field value. The selection model is updated when the field value changes, and the field value is updated when the selection changes.
The selection model uses an identity provider to identify items to account for cases where developers use custom identity providers, for example by overriding `DataProvider.getId`. The implementation currently does not support setting a custom identity provider through data views (`AbstractDataView.setIdentityProvider`), this will be covered in a later change after the Flow team provides a way to get notified when the identity provider changes.

Part of https://github.com/vaadin/platform/issues/2961

### CODE REVIEW

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBoxSelectionModel.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
+package com.vaadin.flow.component.combobox;
```
Comment (web-padawan) [inline vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBoxSelectionModel.java:1]: Let's add the license header to this file.

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/MultiSelectComboBoxMultiSelectIT.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
+package com.vaadin.flow.component.combobox.test;
+
+import com.vaadin.flow.component.combobox.testbench.MultiSelectComboBoxElement;
+import com.vaadin.flow.testutil.TestPath;
+import com.vaadin.testbench.TestBenchElement;
+import com.vaadin.tests.AbstractComponentIT;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openqa.selenium.Keys;
```
Comment (web-padawan) [inline vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/MultiSelectComboBoxMultiSelectIT.java:10]: nit: unused import.

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBoxSelectionModel.java (hunk 1 above)
Comment (sissbruecker) [reply vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBoxSelectionModel.java:16]: Thanks, done.

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/MultiSelectComboBoxMultiSelectIT.java (hunk 2 above)
Comment (sissbruecker) [reply vaadin-combo-box-flow-parent/vaadin-combo-box-flow-integration-tests/src/test/java/com/vaadin/flow/component/combobox/test/MultiSelectComboBoxMultiSelectIT.java]: Removed

**Review by web-padawan — APPROVED** [review]

**Pull request merged**